runtime/error.c

```diff
@@ -181,3 +194,115 @@ TaggedPtr BAL_LANG_ERROR_NAME(message)(TaggedPtr error) {
     ErrorPtr ep = taggedToPtr(error);
     return ep->message;
 }
+
+void _bal_print_mangled_name(const char *mangledName, FILE *fp) {
+    const char *localName;
+    DemangleResult res = demangle(mangledName, &localName, NULL);
+
+    if (res == NON_PUBLIC_BALLERINA_NAME) {
+        fprintf(fp, "%s", localName);
+    }
+    else if (res == PUBLIC_BALLERINA_NAME) {
+        fprintf(fp, "%s (", localName);
+        demangle(mangledName, &localName, fp);
+        fputs(")", fp);
+    }
+    else if (res == NON_BALLERINA_NAME || res == _B_NON_BALLERINA_NAME) {
+        fprintf(fp, "%s", mangledName);
+    }
+}
+
+static DemangleResult demangle(const char *mangledName, const char **localName, FILE *fp) {
+    if (mangledName[0] != '_' || mangledName[1] != 'B') {
+        return NON_BALLERINA_NAME;
+    }
+    mangledName += 2;
+    if (mangledName[0] == '_') {
+        *localName = mangledName + 1;
+        return NON_PUBLIC_BALLERINA_NAME;
+    }
+
+    bool ballerinaOrg = false; 
+    if (mangledName[0] == 'b') {
+        ballerinaOrg = true;
+        mangledName++;
+    }
+
+    const char *org;
+    if (mangledName[0] == '0') {
```

Comment:
Don't need to special case this. You will just end up with an empty org (i.e. *org == '\0')

Reply:
We need this for a case like `_Bb02ioprintln`. If we pass without checking '0' to `demangleCountedName` function, it reads `02` together.

Reply:
Ahh. That's really a design bug on my part.